packages/blade/src/utils/useControllable.ts

```diff
@@ -3,6 +3,17 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 import * as React from 'react';
 
+type ControllableStateSetter<T> = (
+  /**
+   * Sets the state to the given value
```

Comment:
this is that hacky way or we found the main issue?

Reply:
I spent quite a while on that bug even created a [reproduction](https://codesandbox.io/s/usecontrollable-usefirstrender-bug-th7545) of this to debug the issue, could not figure out why this would behave like they way it does. 

There's no point further spending time on this. Because anyways the "hacky" way is much much cleaner than messing around with useRef, useEffect, useisFirstRender, useControllable complexities. 

**Goal:**

So our main goal is that, we don't want to call the onChange when we set the initial value of the tabs. 

**Intent:**

With the "hacky" way the intent is simple: 
_"When setting initial value, don't call onChange"_ 

With the useIsFirstRender the intent becomes:
_"When setting initial value, don't call onChange, and since we set initial value on first render we need to skip calling onChange on the first render"_


Overall the hacky way is much more straightforward than messing with React's lifecycle.